### PR TITLE
New maxRevJobsInAgenda option

### PR DESCRIPTION
**Pull request by vzurauskas**
PR for https://github.com/zerocracy/farm/issues/1906.

- Added a `maxRevJobsInAgenda` option to control max REV jobs in agenda. This includes a new `VsOptionsMaxRevJobs` rule and accessor+mutator in `Options` for the new option.
- Added a `String role(job)` method in `Agenda` to determine the role of a job. This was necessary for the `VsOptionsMaxRevJobs` to be able to count REV jobs in the `Agenda`.
- Updated `Xocument`'s version to enable validations against updated XSD in Datum.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#2110](https://github.com/zerocracy/farm/issues/2110) is now in scope, role is `REV`

**Comment by 0crat** [comment]
Job `gh:zerocracy/farm#2110` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#2110](https://github.com/zerocracy/farm/issues/2110) is now out of scope

**Comment by 0crat** [comment]
Pull request [#2110](https://github.com/zerocracy/farm/issues/2110) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by 0crat** [comment]
Job [#2110](https://github.com/zerocracy/farm/issues/2110) is now in scope, role is `REV`

**Comment by vzurauskas** [comment]
@g4s8 `SyncFarmTest#makesProjectsThreadSafe` keeps randomly timeout'ing for some reason. It sometimes happens on `master` branch too. What can be done about it?

**Comment by g4s8** [comment]
@vzurauskas 
> SyncFarmTest#makesProjectsThreadSafe keeps randomly timeout'ing for some reason.

yes, it's a known issue: CI has one CPU, but this test is submitting a lot of threads, sometimes jobs are killed with TimeoutException by farm classes

File: src/main/groovy/com/zerocracy/stk/pm/staff/elections/elect_performer.groovy [hunk 1]
```diff
@@ -125,6 +125,7 @@ def exec(Project project, XML xml) {
           (wrapped(new VsBigDebt(pmo)))                                          : -100,
           (wrapped(new VsNoRoom(pmo)))                                           : role == 'REV' ? 0 : -100,
           (wrapped(new VsOptionsMaxJobs(pmo)))                                   : role == 'REV' ? 0 : -100,
+          (wrapped(new VsOptionsMaxRevJobs(pmo)))                                : -100,
```
Comment (g4s8) [inline src/main/groovy/com/zerocracy/stk/pm/staff/elections/elect_performer.groovy]: @vzurauskas I think we should use `-100` weight for `REV` jobs only, e.g. if I specify `maxRevJobs = 1`, and get new `DEV` job, and have enough room for new `DEV` job, I'm expecting to be assigned to this `DEV` job. We can change it:
```
role == 'REV' ? -100 : 0
```

Comment (vzurauskas) [reply src/main/groovy/com/zerocracy/stk/pm/staff/elections/elect_performer.groovy]: @g4s8 Good point. Fixed.

**Comment by g4s8** [comment]
@vzurauskas one more comment, can you please rename your commits to start with issue number:
`#1906 - <some message>` so we'll be able to identify it with `git blame` later

**Comment by vzurauskas** [comment]
@g4s8 Done.

**Comment by g4s8** [comment]
@vzurauskas thanks

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:zerocracy/farm#2110` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#2110](https://github.com/zerocracy/farm/issues/2110) is now out of scope

**Pull request merged**